cookbooks/aws-parallelcluster-config/recipes/directory_service.rb

```diff
@@ -53,16 +53,58 @@
   # Then the sssd.conf file is shared through shared_sssd_conf_path to compute nodes.
   # Only contacting the secret manager from head node avoids giving permission to compute nodes to contact the secret manager.
 
+  # Configure SSSD domain properties
+  domain_properties = {
+    # Mandatory properties that must not be overridden by the user.
+    'id_provider' => 'ldap',
+    'ldap_schema' => 'AD',
+
+    # Mandatory properties that are meant to be set via dedicated cluster config properties,
+    # but that can also be overridden via DirectoryService/AdditionalSssdConfigs.
+    'ldap_uri' => ldap_uri_components.join(','),
+    'ldap_search_base' => ldap_search_base,
+    'ldap_default_bind_dn' => node['cluster']['directory_service']['domain_read_only_user'],
+    'ldap_default_authtok' => shell_out!("aws secretsmanager get-secret-value --secret-id #{node['cluster']['directory_service']['password_secret_arn']} --region #{node['cluster']['region']} --query 'SecretString' --output text").stdout.strip,
```

Comment:
Could you please add a timeout to the shell_out command?
Without the timeout it can remain stucked if for some reason you don't have access to secretsmanager endpoint.

`shell_out` [doesn't seem to support](https://docs.chef.io/infra_language/shelling_out/) the timeout option, so let's use Mixlib::ShellOut instead: https://github.com/chef/mixlib-shellout

Reply:
done in the current revision as a separate commit

Reply:
At the end I changed my mind: it is too risky to introduce this construct in a rush.
I've tested that it does not throw an exception when the shell command fails. So, when you specify a non existing secret, cluster creation proceeds and the user ends up having a wrong sssd config file that breaks the communication with the directory.

Furthermore, I noticed that we have such a risky usage of Mixlib::ShellOut in other parts of our cookbook, thus introducing the risk of not failing when the command actually fails.

My proposal is to delay this improvement to the next iteration and to broaden it to every execution we currently have in our cookbook.